extensions/cornerstone/src/commandsModule.ts

```diff
@@ -548,16 +571,28 @@ function commandsModule({
     },
     setViewportColormap: ({ viewportId, displaySetInstanceUID, colormap, immediate = false }) => {
       const viewport = cornerstoneViewportService.getCornerstoneViewport(viewportId);
-
       const actorEntries = viewport.getActors();
 
-      const actorEntry = actorEntries.find(actorEntry => {
-        return actorEntry.uid.includes(displaySetInstanceUID);
-      });
-
-      const { actor: volumeActor, uid: volumeId } = actorEntry;
+      if (viewport instanceof StackViewport) {
+        const actorEntry = actorEntries.find(actorEntry => {
+          return actorEntry.uid.includes(viewportId);
+        });
+        const { actor: volumeActor, uid: volumeId } = actorEntry;
+        viewport.setProperties({ colormap, volumeActor }, volumeId);
+      }
 
-      viewport.setProperties({ colormap, volumeActor }, volumeId);
+      if (viewport instanceof VolumeViewport) {
+        if (!displaySetInstanceUID) {
+          const { viewports } = viewportGridService.getState();
+          // this is not good enough, we need to find the correct displaySetInstanceUID
+          displaySetInstanceUID = viewports.get(viewportId)?.displaySetInstanceUIDs[0];
+        }
+        const actorEntry = actorEntries.find(actorEntry => {
+          return actorEntry.uid.includes(displaySetInstanceUID);
+        });
+        const { actor: volumeActor, uid: volumeId } = actorEntry;
+        viewport.setProperties({ colormap, volumeActor }, volumeId);
+      }
```

Comment:
refactor this since it is duplicated 

```
const setViewportProperties = (viewport, uid) => {
  const actorEntry = actorEntries.find(entry => entry.uid.includes(uid));
  const { actor: volumeActor, uid: volumeId } = actorEntry;
  viewport.setProperties({ colormap, volumeActor }, volumeId);
};
```

Reply:
Done